src/index.tsx

```diff
@@ -1,58 +1,58 @@
 import * as React from 'react';
-import { getBase64, checkResolution } from './utils';
+import { triggerOpenDialog, getListFile, getAcceptString } from './utils';
+import {
+  isResolutionValid,
+  isImageValid,
+  maxFileSizeValid,
+  acceptTypeValid,
+  maxNumberValid,
+} from './validate';
 import {
   ImageType,
   ImageListType,
   ImageUploadingPropsType,
   ErrorsType,
   ResolutionType,
 } from './typings';
+import {
+  DEFAULT_NULL_INDEX,
+  INIT_ERROR,
+  INIT_MAX_NUMBER,
+  DEFAULT_DATA_URL_KEY,
+} from './constants';
 
-const { useRef, useState, useCallback } = React;
-
-const defaultErrors: ErrorsType = {
-  maxFileSize: false,
-  maxNumber: false,
-  acceptType: false,
-  resolution: false,
-};
-const defaultNullIndex = -1;
+const { useRef, useState, useCallback, useMemo } = React;
 
-const ImageUploading: React.FC<ImageUploadingPropsType> = ({
-  multiple = false,
+const ReactImageUploading: React.FC<ImageUploadingPropsType> = ({
+  value = [],
   onChange,
-  maxNumber = 1000,
+  onError,
   children,
-  value = [],
+  dataURLKey = DEFAULT_DATA_URL_KEY,
+  multiple = false,
+  maxNumber = INIT_MAX_NUMBER,
   acceptType,
   maxFileSize,
   resolutionWidth,
   resolutionHeight,
   resolutionType,
-  onError,
-  dataURLKey = 'dataURL',
 }) => {
   const inputRef = useRef<HTMLInputElement>(null);
-  const [keyUpdate, setKeyUpdate] = useState<number>(defaultNullIndex);
-  const [errors, setErrors] = useState<ErrorsType>({ ...defaultErrors });
+  const [keyUpdate, setKeyUpdate] = useState<number>(DEFAULT_NULL_INDEX);
+  const [errors, setErrors] = useState<ErrorsType>(null);
   const [isDragging, setIsDragging] = useState<Boolean>(false);
 
-  const handleClickInput = useCallback((): void => {
-    inputRef.current && inputRef.current.click();
-  }, [inputRef]);
+  const handleClickInput = useCallback(() => triggerOpenDialog(inputRef), [
+    inputRef,
+  ]);
 
   const onImageUpload = useCallback((): void => {
-    setKeyUpdate((prevKey) => {
-      if (prevKey >= 0) {
-        return defaultNullIndex;
-      }
-      return prevKey;
-    });
+    setKeyUpdate(DEFAULT_NULL_INDEX);
     handleClickInput();
   }, [handleClickInput]);
 
   const onImageRemoveAll = useCallback((): void => {
-    onChange && onChange([]);
+    if (onChange) onChange([]);
```

Comment:
You can use something like this to make look more concise
```suggestion
onChange && onChange([]);
```